tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -1077,18 +1107,24 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
             implicit val env = env0
             extractLet { newLhs =>
               js.If(transformExpr(newCond),
-                  pushLhsInto(newLhs, thenp), pushLhsInto(newLhs, elsep))
+                  pushLhsInto(newLhs, thenp, tailPosLabels),
+                  pushLhsInto(newLhs, elsep, tailPosLabels))
             }
           }
 
         case Try(block, errVar, handler, finalizer) =>
           extractLet { newLhs =>
+            val blockAndHandlerTailPosLabels =
+              if (finalizer == EmptyTree) tailPosLabels
+              else Set.empty[String]
```

Comment:
Are you 100% positive you need this?

Consider:
```javascript
while (true) { try { break; } finally { console.log("finally"); } } // prints "finally"
```

Needing this would make JavaScript an even more broken language :P